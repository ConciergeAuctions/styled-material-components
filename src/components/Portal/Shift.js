// @flow
import React, { Component, type Node } from "react";

type ShiftPropsType = {|
  children: Node,
  className: string,
  direction: string,
  onClick: (e: SyntheticMouseEvent<HTMLDivElement>) => void,
  open?: boolean // TODO which ones?
|};

export class Shift extends Component<ShiftPropsType> {
  componentDidMount() {
    document.body.classList.add("shift");
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.open && !this.props.open) {
      document.body.classList.add(this.props.direction);
    } else if (!nextProps.open && this.props.open) {
      document.body.classList.remove(this.props.direction);
    }
  }

  componentWillUnmount() {
    document.body.classList.remove("shift");
  }

  render() {
    return <div className={this.props.className}>{this.props.children}</div>;
  }
}
