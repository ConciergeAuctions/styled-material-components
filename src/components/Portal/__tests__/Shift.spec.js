import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Shift } from '../Shift';

import {
  STYLED_COMPONENTS_CONTEXT,
  STYLED_COMPONENTS_CHILD_CONTEXT_TYPES,
} from '../../../../test/utils';

describe('Shift', () => {
  let mounted;
  it('mounts', () => {
    mounted = shallow(<Shift className="testclass">Content stuff!</Shift>, {
      context: { ...STYLED_COMPONENTS_CONTEXT },
      childContextTypes: { ...STYLED_COMPONENTS_CHILD_CONTEXT_TYPES },
    });
    expect(mounted).toBeTruthy();
  });

  it('matches snapshot', () => {
    expect(toJSON(mounted)).toMatchSnapshot();
  });

  it.skip('renders its children as content', () => {});

  it.skip('adds a "shift" class to the document body on mount', () => {});

  it.skip('when opened, adds its direction as a class  to the document body', () => {});

  it.skip('when closed, removes its direction class', () => {});

  it.skip('removes the "shift" class from the document body on unmount', () => {});
});
