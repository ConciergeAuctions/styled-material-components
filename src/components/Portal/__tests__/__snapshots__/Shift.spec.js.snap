// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shift matches snapshot 1`] = `
<div
  className="testclass"
>
  Content stuff!
</div>
`;
